src/passes/Print.cpp

```diff
@@ -35,12 +35,7 @@ static bool isFullForced() {
 }
 
 static std::ostream& printName(Name name, std::ostream& o) {
-  // we need to quote names if they have tricky chars
```

Comment:
What about the case where the name has not arrived from the binary reading code? I think we may still need to escape there.

Reply:
based on irc discussion, I think we still need quoting here - we need to emit something we can parse.

we should probably check if we need to quote, as escaping may have removed the problem for us. if not, then we must quote.